src/cargo/ops/cargo_package/mod.rs

```diff
@@ -160,11 +160,8 @@ fn create_package(
     }
 
     let filename = pkg.package_id().tarball_name();
-    let dir = ws.target_dir().join("package");
-    let mut dst = {
-        let tmp = format!(".{}", filename);
-        dir.open_rw_exclusive_create(&tmp, gctx, "package scratch space")?
-    };
+    let dir = ws.build_dir().join("package").join("tmp-crate");
```

Comment:
I am a bit unsure about adding this `tmp-crate`. From [the doc](https://doc.rust-lang.org/1.88.0/cargo/reference/build-cache.html)


> Contains the output of the [cargo package](https://doc.rust-lang.org/1.88.0/cargo/commands/cargo-package.html) and [cargo publish](https://doc.rust-lang.org/1.88.0/cargo/commands/cargo-publish.html) commands.

which means something is there, and it was not clear whether they were intermediate or final artifacts. To minimize the impact, is there a way to do this without the new `tmp-crate` directory?

Reply:
We already have `tmp-registry` and the decompressed directories in there.

Is the concern about mixing intermediate and final artifacts in similar locations?  We can split that but since its larger than just this, should we split this out?

Reply:
The concern is people already relying on the package directory layout with `cargo publish`. While that is not intended, the existing doc putting package and publish together makes them looks like the same. Should we give people one more release to adjust?
We could include a note in the changelog saying this will change in the next release even you are not using `build.build-dir`. What do you think?